compiler/rustc_hir/src/definitions.rs

```diff
@@ -155,6 +155,32 @@ pub struct DisambiguatedDefPathData {
     pub disambiguator: u32,
 }
 
+impl DisambiguatedDefPathData {
+    pub fn format<W: Write>(&self, writer: &mut W, verbose: bool) -> fmt::Result {
+        match self.data.name() {
+            DefPathDataName::Named(name) => {
+                if Ident::with_dummy_span(name).is_raw_guess() {
+                    writer.write_str("r#")?;
+                }
```

Comment:
Hmm should we maybe handle this in other cases too? Seems a bit odd to have the `fmt::Display` impl for `DefPathData` exist while not handling that aspect.

Reply:
I CAN add this check also in `impl Display for DefPathData` but I am concerned that it would introduce a (breacking?) change for the legacy names?

Reply:
Do any tests change behavior when you do that? `is_raw_guess` should only return `true` for keywords, so at least this should be rare.

Reply:
Adding this to `DefPathData` is making a _lot of_ tests fail to compile with `undefined reference` errors. Most of them are for:
```
   = note: /bin/ld: /home/adema/Sources/rust/build/x86_64-unknown-linux-gnu/test/ui/threads-sendsync/task-comm-13/a.task_comm_13.7rcbfp3g-cgu.7.rcgu.o: in function `core::ops::function::FnOnce::call_once{{vtable-shim}}':
           task_comm_13.7rcbfp3g-cgu.7:(.text._ZN4core3ops8function6FnOnce40call_once$u7b$$u7b$vtable.shim$u7d$$u7d$17h07af120783683994E+0xef): undefined reference to `std::panicking::try::cleanup'
           collect2: error: ld returned 1 exit status
```
or similar.

Reply:
Ah, weird, guess it's fine then.

Reply:
Though we could go the other way and move that check to `ty::print::pretty`, which might be the only place that cares.

Reply:
I moved it, should be all good now!